python/cudf/cudf/_lib/sort.pyx

```diff
@@ -145,6 +146,53 @@ def order_by(list columns_from_table, object ascending, str na_position):
     return Column.from_unique_ptr(move(c_result))
 
 
+@acquire_spill_lock()
+def sort(
+    list values,
+    list column_order=None,
+    list null_precedence=None,
+):
+    """
+    Sort the table in ascending/descending order.
+
+    Parameters
+    ----------
+    values : list[Column]
+        Columns of the table which will be sorted
+    column_order : list[bool], optional
+        Sequence of boolean values which correspond to each column in
+        keys providing the sort order (default all True).
+        With True <=> ascending; False <=> descending.
+    null_precedence : list[str], optional
+        Sequence of "first" or "last" values (default "first")
+        indicating the position of null values when sorting the keys.
+    """
+    cdef table_view values_view = table_view_from_columns(values)
+    cdef vector[order] c_column_order
+    cdef vector[null_order] c_null_precedence
+    cdef unique_ptr[table] result
+    ncol = len(values)
+    column_order = column_order or [True] * ncol
+    null_precedence = null_precedence or ["first"] * ncol
+    for asc, null in zip(column_order, null_precedence):
+        c_column_order.push_back(order.ASCENDING if asc else order.DESCENDING)
+        if asc ^ (null == "first"):
+            c_null_precedence.push_back(null_order.AFTER)
+        elif asc ^ (null == "last"):
+            c_null_precedence.push_back(null_order.BEFORE)
+        else:
+            raise ValueError(f"Invalid null precedence {null}")
```

Comment:
This bit of code is now replicated three times in this file, is it time to factor it out? (`order_by` doesn't take an iterable of `null_precedences`, but one could use `zip(column_order, repeat(na_position))` in that case.

Reply:
Yes, I'm a strong +1 for refactoring this. I had considered doing so but hadn't assessed it fully. I'll work on that.

Reply:
@wence- Can we temporarily skip this refactor and do it in your work-in-progress branch?

Reply:
@wence- I added commit f2eef2022edd3bdc8b22e6fb45f94d9d329b9778 which adapts your changes.